apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs

```diff
@@ -36,5 +58,251 @@ public void CreateConnection()
+                    "TestTable", new SpannerParameterCollection
+                    {
+                        {"Key", SpannerDbType.String, keys[0]},
+                        {"Int64Value", SpannerDbType.Int64, 0L}
+                    });
+                await updateCmd.ExecuteNonQueryAsync();
+
+                // Delete row for keys[1]
+                var deleteCmd = connection.CreateDeleteCommand(
+                    "TestTable", new SpannerParameterCollection
+                    {
+                        {"Key", SpannerDbType.String, keys[1]}
+                    });
+                await deleteCmd.ExecuteNonQueryAsync();
+            }
+            // End sample
+        }
+
+        // Sample: SpannerFaultDetectionStrategy
+        private class SpannerFaultDetectionStrategy : ITransientErrorDetectionStrategy
+        {
+            /// <inheritdoc />
+            public bool IsTransient(Exception ex) => ex.IsTransientSpannerFault();
+        }
+        // End sample
+
+        [Fact]
+        public async Task TransactionAsync()
+        {
+            await _fixture.EnsureTestDatabaseAsync().ConfigureAwait(false);
+
+            // Sample: TransactionAsync
+            // Additional: BeginTransactionAsync
+            var retryPolicy = new RetryPolicy<SpannerFaultDetectionStrategy>(RetryStrategy.DefaultExponential);
+
+            await retryPolicy.ExecuteAsync(
+                async () =>
+                {
+                    using (var connection =
+                        new SpannerConnection(
+                            $"Data Source=projects/{_projectId}/instances/{_instanceName}/databases/{_databaseName}"))
+                    {
+                        await connection.OpenAsync();
+
+                        using (var transaction = await connection.BeginTransactionAsync())
+                        {
+                            var cmd = connection.CreateInsertCommand(
+                                "TestTable", new SpannerParameterCollection
+                                {
+                                    {"Key", SpannerDbType.String},
+                                    {"StringValue", SpannerDbType.String},
+                                    {"Int64Value", SpannerDbType.Int64}
+                                });
+                            cmd.Transaction = transaction;
+
+                            // This executes a single transactions with alls row written at once during CommitAsync().
+                            // If a transient fault occurs, this entire method is re-run.
+                            for (var i = 0; i < 5; i++)
+                            {
+                                cmd.Parameters["Key"].Value = Guid.NewGuid().ToString("N");
+                                cmd.Parameters["StringValue"].Value = $"StringValue{i}";
+                                cmd.Parameters["Int64Value"].Value = i;
+                                await cmd.ExecuteNonQueryAsync();
```

Comment:
What does this return when it's not committing at the same time? Always 1, on the grounds that it assumes it *will* insert later?

Reply:
it always should return number of rows affected even if it doesn't commit at that time.
Seeing how its built into ADO.NET, I don't feel its super important to highlight -- as its also very reasonable to ignore the result and rely on an exception not being thrown.

Reply:
Sure, not a problem.